File: src/internalStructure.js
const data = require ('./tables/internalStructure.js').table;
const round = require('./helpers/calculations').round

module.exports = {
    getTable: (baseWeight) => {
        return data[baseWeight];
    },
    calculateWeight: (baseWeight, endoSteel) => {
        endoSteel || (endoSteel = false);
        if (endoSteel) {
            return round((baseWeight /10)/2, 0.5 );
        } else {
            return (baseWeight /10 )
        }
    }
}

File: src/internal_structure.py
#p.py
import math

class InternalStructure(object):
    @staticmethod
    def calculate(mechTonnage, endo_steel = False):
        if endo_steel:  
            return math.ceil((mechTonnage/10)/2 *2)/2
        else:
            return mechTonnage/10 
    @staticmethod
    def get_table(mechTonnage):
        
        
        return is_table[mechTonnage]
